packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -43,7 +43,7 @@ class CurlyComponentManager {
   create(definition, args, dynamicScope, hasBlock) {
     let parentView = dynamicScope.view;
 
-    let klass = definition.ComponentClass;
+    let klass = definition.getComponentClass(args);
```

Comment:
Because for input, class selection is dynamic: https://github.com/emberjs/ember.js/pull/13615/files#diff-f9ead5f5a3d079b17b0d5b74759b00b0R9